controller/array_action/array_mediator_svc.py

```diff
@@ -468,26 +473,19 @@ def _stop_and_delete_fcmap(self, fcmap_id):
         self._stop_fcmap(fcmap_id)
         self._delete_fcmap(fcmap_id, force=True)
 
-    def _delete_all_fcmaps_as_source_if_exist(self, snapshot_name):
-        snapshot_as_source_fcmaps = self._get_fcmaps(snapshot_name, ENDPOINT_TYPE_SOURCE)
-        unfinished_fcmaps = [fcmap.name for fcmap in snapshot_as_source_fcmaps
-                             if fcmap.status != FCMAP_STATUS_DONE]
-        if unfinished_fcmaps:
-            raise controller_errors.ObjectIsStillInUseError(snapshot_name, unfinished_fcmaps)
-        for fcmap in snapshot_as_source_fcmaps:
-            self._delete_fcmap(fcmap.id, force=False)
-
     def _delete_snapshot(self, cli_volume):
-        snapshot_name = cli_volume.name
-        fcmap = self._get_fcmap_as_target_if_exists(snapshot_name)
-        if not fcmap:
-            raise controller_errors.ObjectNotFoundError(snapshot_name)
-
-        fcmap_id = cli_volume.FC_id
-        if fcmap_id == 'many':
-            self._delete_all_fcmaps_as_source_if_exist(snapshot_name)
-        self._stop_and_delete_fcmap(fcmap.id)
-        self._delete_volume_by_name(snapshot_name)
+        self._delete_object(cli_volume.name, snapshot=True)
+
+    def _delete_object(self, object_name, snapshot=False):
```

Comment:
why is there a default value? and why that value?
why not use the generic object_type (`*_TYPE_NAME`)?

Reply:
cause I dont care for the "volume" option only the snapshot

Reply:
ok

Reply:
then could you just rename it to `is_snapshot`?

Reply:
done